lib/seattleflu/id3c/cli/command/etl/redcap_det_asymptomatic_swab_n_send.py

```diff
@@ -43,11 +44,15 @@
     "asymptomatic-swab-n-send",
     redcap_url = REDCAP_URL,
     project_id = PROJECT_ID,
-    required_instruments = REQUIRED_INSTRUMENTS,
     revision = REVISION,
     help = __doc__)
 
-def redcap_det_asymptomatic_swab_n_send(*, db: DatabaseSession, cache: TTLCache, det: dict, redcap_record: dict) -> Optional[dict]:
+@get_first_redcap_record_instance_only
```

Comment:
Does the `@get_first_redcap_record_instance_only` decorator resolve before the `@redcap_det.command_for_project`? If so, does the behavior around required instruments differ across ETLs still? If the behavior will consistently just to iterate over all the record instances that exist (since its already reduced to first instance only for non-longitudinal ETLs) to check for required instruments, I'm wondering if this could still be in core.

Reply:
`@get_first_redcap_record_instance_only` is injected between `@redcap_det.command_for_project` and the custom ETL function. 

The check for required instruments got moved to customizations because it wouldn't work the same way for repeating REDCap projects. In repeating projects, each record instance has all of the fields, but only has the values for that specific instance. So you can't just check the first instance for all required instruments.